contracts/stake/src/state.rs

```diff
@@ -70,204 +70,219 @@ impl StakeState {
     pub fn stake(&mut self, stake: Stake) {
         self.check_new_block();
 
-        if stake.value < MINIMUM_STAKE {
-            panic!("The staked value is lower than the minimum amount!");
-        }
+        let payload = stake.payload();
+        let signature = stake.signature();
 
-        // allot a stake to the given key and increment the signature counter
-        let loaded_stake = self.load_or_create_stake_mut(&stake.public_key);
+        let loaded_stake = self.load_or_create_stake_mut(&payload.account);
 
-        let counter = loaded_stake.counter();
+        // ensure the stake is at least the minimum and that there isn't an
+        // amount staked already
+        if payload.value < MINIMUM_STAKE {
+            panic!("The staked value is lower than the minimum amount!");
+        }
 
-        loaded_stake.increment_counter();
-        loaded_stake.insert_amount(stake.value, rusk_abi::block_height());
+        if loaded_stake.amount.is_some() {
```

Comment:
The reason we don't do the above is because an account that's not staking can still have a reward. This should be taken in tandem with splitting rewards from stakes, as per above.